master/pkg/syncx/queue/queue.go

```diff
@@ -0,0 +1,57 @@
+package queue
+
+import (
+	"sync"
+)
+
+// Queue is a thread-safe queue.
+type Queue[T any] struct {
+	mu      sync.Mutex
+	cond    *sync.Cond // used to wait for elements in the queue
+	elems   []T
+}
+
+// New creates a new queue.
+func New[T any]() *Queue[T] {
+	q := &Queue[T]{elems: make([]T, 0)}
+	q.cond = sync.NewCond(&q.mu)
+	return q
+}
+
+// Put adds an element to the queue. If the queue was configured with a maximum size, then Put will
+// block until space is available.
+func (q *Queue[T]) Put(t T) {
+	q.mu.Lock()
+	defer q.mu.Unlock()
+
+	if q.empty() {
+		q.cond.Broadcast()
+	}
+	q.elems = append(q.elems, t)
+}
+
+// Get removes and returns an element from the queue. If the queue is empty, then Get will block
+// until an element is available.
+func (q *Queue[T]) Get() T {
```

Comment:
While it's a longer name, I have a mild preference for using the idiomatic operation name of `Dequeue`. Some queues have a "peek" operation and it'd be nice if our naming could save users of the queue a trip to the docs by disambiguating which operation this is.

Reply:
I just replicated the python queue api